src/apiQueries/useHorizonNetworkInfo.ts

```diff
@@ -0,0 +1,23 @@
+import { useQuery } from "@tanstack/react-query";
+import { HORIZON_URL } from "@/constants/envVariables";
+import { fetchStellarApi } from "@/helpers/fetchStellarApi";
+import { AppError } from "@/types";
+
+export type HorizonNetworkInfo = {
+  network_passphrase: string;
+};
+
+export const useHorizonNetworkInfo = () => {
+  const query = useQuery<HorizonNetworkInfo, AppError>({
+    queryKey: ["horizon", "network", HORIZON_URL],
+    queryFn: async () => {
+      return await fetchStellarApi(`${HORIZON_URL}/`, undefined, {
+        notFoundMessage: "Network information could not be retrieved.",
+      });
+    },
+    staleTime: Infinity, // no need to refetch unless redeployed
+    gcTime: Infinity,
```

Comment:
I know that it's unlikely in the current scenario not to have `HORIZON_URL`, but it might be good to add `enabled` just in case it is used somewhere else in the future.

```suggestion
    gcTime: Infinity,
    enabled: Boolean(HORIZON_URL),
```